src/main/kotlin/dev/encore/intellij/StartupActivity.kt

```diff
@@ -0,0 +1,46 @@
+package dev.encore.intellij
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.startup.StartupActivity
+import com.intellij.util.text.SemVer
+import dev.encore.intellij.sqldb.Detector
+import dev.encore.intellij.utils.isInEncoreApp
+import dev.encore.intellij.utils.runEncoreCommand
+
+/*
+ * Startup Activity allows us to run blocking external calls to the encore daemon to extract information
+ * about the project, which we can then store and reference at runtime
+ */
+class StartupActivity : StartupActivity.Background, StartupActivity.RequiredForSmartMode {
+    override fun runActivity(project: Project) {
+        // Skip our startup if it's not an Encore app
+        if (!isInEncoreApp(project)) {
+            return
+        }
+
+        Encore.LOG.info("Performing startup activity for Encore...")
+
+        // Verify Encore is installed
+        val versionOutput = runEncoreCommand(project, "version") ?: return
+        if (!versionOutput.startsWith("encore version ")) {
+            return
+        }
+        Encore.encoreInstalled = true
+
+        // Get the version
+        val versionString = versionOutput.trim().lines()[0].removePrefix("encore version ")
+        if (versionString.startsWith("v")) {
```

Comment:
Maybe we need `encore version --json` or similar :) but good enough for now

Reply:
In the future yeah :)